integreat_cms/google_translate_api/apps.py

```diff
@@ -0,0 +1,128 @@
+        try:
+            google_translator_v2 = translate_v2.Client(credentials=credentials)
+            # Version 2 (Basic) does not distinguish source and target language
+            supported_languages = google_translator_v2.get_languages()
+            self.supported_source_languages = []
+            self.supported_target_languages = []
+
+            for language in supported_languages:
+                self.supported_source_languages += [language["language"]]
+                self.supported_target_languages += [language["language"]]
+
+            assert self.supported_source_languages
+            logger.debug(self.supported_source_languages)
+            assert self.supported_target_languages
+            logger.debug(self.supported_target_languages)
+
+            logger.info("Google Translate API is enabled.")
+        except AssertionError as e:
+            logger.error(e)
+            logger.error("Google translate is not available")
+
+    def ready_v3(self, credentials: Credentials) -> None:
+        """
+        Preparing Google Translate with advanced version
+        """
+        logger.info(translate_v3)
+        try:
+            google_translator_v3 = translate_v3.TranslationServiceClient(
+                credentials=credentials
+            )
+            parent = settings.GOOGLE_PARENT_PARAM
+
+            supported_languages = google_translator_v3.get_supported_languages(
+                parent=parent
+            ).languages
+
+            self.supported_source_languages = []
+            self.supported_target_languages = []
+
+            for language in supported_languages:
+                if language.support_source:
+                    self.supported_source_languages += [language.language_code]
+                if language.support_target:
+                    self.supported_target_languages += [language.language_code]
+
+            assert self.supported_source_languages
+            logger.debug(
+                "Supported source languages by Google Translate: %r",
+                self.supported_source_languages,
+            )
+            assert self.supported_target_languages
+            logger.debug(
+                "Supported target languages by Google Translate: %r",
+                self.supported_target_languages,
+            )
+
+            logger.info("Google Translate API is enabled.")
+        except AssertionError as e:
+            logger.error(e)
+            logger.error("Google translate is not available")
+
+    def ready(self) -> None:
+        """
+        Checking if API is available
+        """
+        # Only check availability if running a server
+        if "runserver" in sys.argv or "APACHE_PID_FILE" in os.environ:
+            if settings.GOOGLE_TRANSLATE_ENABLED:
+                try:
+                    credentials = service_account.Credentials.from_service_account_file(
+                        settings.GOOGLE_APPLICATION_CREDENTIALS
+                    )
```

Comment:
It looks like you should be able to use `Credentials.from_service_account_info()` instead, supplying a `dict` of essentially what should be in the file otherwise. Man, the Google documentation is weird and frustrating concerning this… But I wager ["in Google format"](https://googleapis.dev/python/google-auth/latest/_modules/google/auth/jwt.html#OnDemandCredentials.from_service_account_info) means exactly what the file is.

Reply:
Seems like this is the case, I found an example here: https://googleapis.dev/python/google-auth/latest/user-guide.html#service-account-private-key-files

Reply:
But the private key has to be stored in the file anyway, doesn't it?

Reply:
> Credentials.from_service_account_info()

I understood so too.

Reply:
> But the private key has to be stored in the file anyway?

I understood so too.